### PR TITLE
fix: remove stat files and add count files for scout upload VCFs

### PR DESCRIPTION
**Pull request by khurrammaqbool**
### This PR:
Cleans up vep rules and adds stats files for all endpoint VCFs to be useful for automatic uploads of VCF files. 

Added: *_counts files for each end point VCF file including *.all.filtered.pass.vcf.gz 
Removed: *_stats files from vep rules 

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

**Comment by khurrammaqbool** [comment]
@ivadym , I added stats files for all the filtered.passed VCF files and also added varcaller names to the output files.

**Comment by khurrammaqbool** [comment]
@ashwini06 and @ivadym, bcftools counts is added for all end point VCF files and bcftools stats is removed from vep rules. 
This pull request is ready for review!

**Review by ashwini06 — COMMENTED** [review]
I added the comments about the count files for which we are not uploading to scout. It's up to you to either address it here or @ivadym if would like to take it in his upcoming PR. You both can decide on it :) 

 I wonder if this `.all.filtered.pass.stats` are correctly generated. Check with some dry runs. If not include the required file in https://github.com/Clinical-Genomics/BALSAMIC/blob/164e3aec27af8061bd5d618e048c8f970a120e77/BALSAMIC/workflows/balsamic.smk#L262

**Comment by khurrammaqbool** [comment]
@keyvanelhami could you please provide the updated list of end-point VCF files to be uploaded to scout we can use to generate counts for?

**Comment by keyvanelhami** [comment]
@khurrammaqbool We are using below flags and files for Scout upload. You can also look into an example of a yaml file for Scout: `/home/proj/production/housekeeper-bundles/ablepoodle/2022-04-07/scout_load.yaml `

```
vcf_cancer: /home/proj/production/housekeeper-bundles/ablepoodle/2022-04-07/SNV.somatic.ablepoodle.vardict.all.filtered.pass.vcf.gz
vcf_cancer_sv: /home/proj/production/housekeeper-bundles/ablepoodle/2022-04-07/SV.somatic.ablepoodle.manta.all.filtered.pass.vcf.gz
```

**Comment by khurrammaqbool** [comment]
> /home/proj/production/housekeeper-bundles/ablepoodle/2022-04-07/scout_load.yaml

The list seems to include manta. We have decided to upload svdb instead. Isn't it an old list?

**Comment by keyvanelhami** [comment]
> have decided to upload svdb instead. Isn't it an old list?

But is the svdb PR merge and deployed in production? If so we should open an issue so `cg` points to the svdb vcf instead

**Comment by khurrammaqbool** [comment]
> > have decided to upload svdb instead. Isn't it an old list?
> 
> But is the svdb PR merged and deployed in production? If so we should open an issue so `cg` points to the svdb vcf instead

It will be merged in develop and is linked to the delivery of files @ivadym is currently working with.

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule [hunk 1]
```diff
@@ -74,9 +78,11 @@ bcftools view {input.vcf} \
 
 tabix -p vcf -f {output.vcf_filtered};
 
-bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
+bcftools view -f PASS -o {output.vcf_pass_tnhaplotyper} -O z {output.vcf_filtered};
+
+tabix -p vcf -f {output.vcf_pass_tnhaplotyper};
 
-tabix -p vcf -f {output.vcf_pass};
+bcftools +counts {output.vcf_pass_tnhaplotyper} > {output.bcftools_counts};
```
[the remark this answers is not in the document]
Comment (khurrammaqbool) [reply BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_normal.rule]: fixed!

File: BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule [hunk 2]
```diff
@@ -72,9 +76,11 @@ bcftools view {input.vcf} \
 
 tabix -p vcf -f {output.vcf_filtered};
 
-bcftools view -f PASS -o {output.vcf_pass} -O z {output.vcf_filtered};
+bcftools view -f PASS -o {output.vcf_pass_tnhaplotyper} -O z {output.vcf_filtered};
+
+tabix -p vcf -f {output.vcf_pass_tnhaplotyper};
 
-tabix -p vcf -f {output.vcf_pass};
+bcftools +counts {output.vcf_pass_tnhaplotyper} > {output.bcftools_counts};
```
[the remark this answers is not in the document]
Comment (khurrammaqbool) [reply BALSAMIC/snakemake_rules/annotation/varcaller_filter_tumor_only.rule]: fixed!

File: BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_normal.rule [hunk 3]
```diff
@@ -73,7 +77,9 @@ bcftools view -f PASS --threads {threads} --regions-file {input.wgs_calling_file
 
 tabix -p vcf -f {output.vcf_filtered};
 
-bcftools view -f PASS --threads {threads} -O z -o {output.vcf_pass} {output.vcf_filtered}
+bcftools view -f PASS --threads {threads} -O z -o {output.vcf_pass_tnhaplotyper} {output.vcf_filtered}
+
+tabix -p vcf -f {output.vcf_pass_tnhaplotyper}
 
-tabix -p vcf -f {output.vcf_pass}
+bcftools +counts {output.vcf_pass_tnhaplotyper} > {output.bcftools_counts};
```
[the remark this answers is not in the document]
Comment (khurrammaqbool) [reply BALSAMIC/snakemake_rules/annotation/varcaller_wgs_filter_tumor_normal.rule]: fixed!

File: CHANGELOG.rst [hunk 4]
```diff
@@ -29,6 +30,7 @@ Fixed:
 * Automate balsamic version for readthedocs install page #888
 * ``collect_qc_metrics.py`` failing for WGS cases with empty ``capture_kit`` argument #850
 * QC metric validation for different panel bed version #855
+* bcftools_stats from vep #898
```
Comment (ivadym) [inline CHANGELOG.rst:33]: Should this be included inside the `Removed` section?

**Review by ivadym — APPROVED** [review]
Looks good to me 👍

File: CHANGELOG.rst (hunk 4 above)
Comment (khurrammaqbool) [reply CHANGELOG.rst]: Thanks @ivadym. Fixed!

**Pull request merged**